node/src/gql/mutation/login.ts

```diff
@@ -0,0 +1,53 @@
+/*
+This software is Copyright ©️ 2020 The University of Southern California. All Rights Reserved. 
+Permission to use, copy, modify, and distribute this software and its documentation for educational, research and non-profit purposes, without fee, and without a written agreement is hereby granted, provided that the above copyright notice and subject to the full license file found in the root of this software deliverable. Permission to make commercial use of this software may be obtained by contacting:  USC Stevens Center for Innovation University of Southern California 1150 S. Olive Street, Suite 2300, Los Angeles, CA 90115, USA Email: accounting@stevens.usc.edu
+
+The full terms of this copyright and license should always be found in the root directory of this software deliverable as "license.txt" and if these terms are not found with this software, please contact the USC Stevens Center for the full license.
+*/
+import axios from 'axios';
+import { GraphQLString, GraphQLObjectType } from 'graphql';
+import UserType from 'gql/types/user';
+import { User } from 'models/User';
+import { User as UserSchema } from 'models';
+
+export const login = {
+  type: UserType,
+  args: {
+    accessToken: { type: GraphQLString },
+  },
+  resolve: async (
+    _root: GraphQLObjectType,
+    args: { accessToken: string }
+  ): Promise<User> => {
+    if (!args.accessToken) {
+      throw new Error('missing required param accessToken');
+    }
+    try {
+      const endpoint = `https://www.googleapis.com/oauth2/v1/userinfo?alt=json&access_token=${args.accessToken}`;
+      const response = await axios.get(endpoint);
+      const user = await UserSchema.findOne({ email: response.data.email });
```

Comment:
Probably better to have a field `User.googleId` which has a sparse index (null by default) and populate/authenticate against that. The reason I favor it, is it may save us from having to do another db migration in the not-too-distant future (when we add additional authentication options for users)